libgadget/gravpm.c

```diff
@@ -43,34 +39,40 @@ static PetaPMFunctions functions [] =
 
 static PetaPMGlobalFunctions global_functions = {NULL, NULL, potential_transfer};
 
-static PetaPM pm[1];
 static PetaPMRegion * _prepare(PetaPM * pm, void * userdata, int * Nregions);
 
-void gravpm_init_periodic() {
-    petapm_init(pm, All.BoxSize, All.Nmesh, MPI_COMM_WORLD);
+PetaPM
+gravpm_init_periodic(double BoxSize, double Asmth, int Nmesh) {
```

Comment:
void gravpm_init_periodic(PetaPM *, ...)

Instead of returning a copy of PetaPM object -- consistency with existing calling conventions.